### PR TITLE
Add config support for ssl_version and ssl_cipher_list.

### PR DESCRIPTION
**Pull request by frameworked**
Enforcing TLSv1_2 now_2 now woring.

### CODE REVIEW

File: lib/thin/backends/base.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def threadpool_size=(size)
       def threaded?; @threaded end
             
       # Allow using SSL in the backend.
-      attr_writer :ssl, :ssl_options
+      attr_writer :ssl, :ssl_options, :ssl_version, :ssl_cipher_list
```
Comment (macournoyer) [inline lib/thin/backends/base.rb]: Those two new attr don't seem to be used anywere. Version and cypher_list are stored in `ssl_options`.

File: lib/thin/server.rb [hunk 2]
```diff
@@ -86,7 +86,7 @@ class Server
     def_delegators :backend, :threaded?, :threaded=, :threadpool_size, :threadpool_size=
     
     # Allow using SSL in the backend.
-    def_delegators :backend, :ssl?, :ssl=, :ssl_options=
+    def_delegators :backend, :ssl?, :ssl=, :ssl_options=, :ssl_version=, :ssl_cipher_list=
```
Comment (macournoyer) [inline lib/thin/server.rb]: Same comment as above. Not used.

File: lib/thin/controllers/controller.rb [hunk 3]
```diff
@@ -51,11 +51,13 @@ def start
         server.threaded                       = @options[:threaded]
         server.no_epoll                       = @options[:no_epoll] if server.backend.respond_to?(:no_epoll=)
         server.threadpool_size                = @options[:threadpool_size] if server.threaded?
+        server.ssl_version                    = @options[:ssl_version]
```
Comment (macournoyer) [inline lib/thin/controllers/controller.rb]: Not used

**Comment by tasaif** [comment]
+1

**Comment by frameworked** [comment]
Removed the unused attributes.

**Comment by emilpetkov** [comment]
@macournoyer PR looks good now, after today's cleanup from @frameworked - can we try to merge this? We are also depending on same functionality of thin and TLS

**Comment by macournoyer** [comment]
Thanks a lot! Will release this weekend.

**Comment by macournoyer** [comment]
Done: https://rubygems.org/gems/thin/versions/1.7.2

**Pull request merged**